weather.py

```diff
@@ -1,109 +1,134 @@
-        else:
-            yield key, value
+    data['unit_pressure'] = 'hPa'
+    if unit_temp == 'fahrenheit':
+        data['unit_temperature'] = 'F'
+    else:
+        data['unit_temperature'] = 'C'
+    if unit_speed == 'miles_hour':
+        data['unit_speed'] = 'mph'
+    else:
+        data['unit_speed'] = 'm/s'
+
+    loc = obs.get_location()
+    weather = obs.get_weather()
+
+    data['city'] = loc.get_name()
+    data['country'] = loc.get_country()
+    data['temp'] = round(weather.get_temperature(unit=unit_temp)['temp'])
+    data['text'] = weather.get_detailed_status()
+    data['humidity'] = weather.get_humidity()
+    data['pressure'] = weather.get_pressure()['press']
+
+    wind = weather.get_wind(unit=unit_speed)
+    # Wind direction is sometimes unset; I'm assuming this occurs when
+    # different weather stations for the same location report the wind
+    # blowing in conflicting directions
+    if 'deg' not in wind:
+        wind['deg'] = None
+    data['wind_speed'] = round(wind['speed'])
+    data['wind_direction'] = wind['deg']
+    data['wind_direction_fuzzy'] = fuzzy_direction(wind['deg'])
+    data['wind_direction_arrow'] = arrow_direction(wind['deg'])
+
+    data['sunrise'] = unix_to_hhmm(weather.get_sunrise_time())
+    data['sunset'] = unix_to_hhmm(weather.get_sunset_time())
+    return format_str.format(**data)
+
+def count_set(*args):
+    count = 0
+    for a in args:
+        if a:
+            count += 1
+    return count
 
 if __name__ == '__main__':
     p = argparse.ArgumentParser()
-    p.add_argument('woeid')
     p.add_argument('--format', metavar='F',
-                   default='{city}, {region}: {text}, {temp}\u00b0{unit_temperature}',
+                   default='{city}, {country}: {text}, '
+                           '{temp}\u00b0{unit_temperature}',
                    help="format string for output")
     p.add_argument('--position', metavar='P', type=int, default=-2,
                    help="position of output in JSON when wrapping i3status")
-    p.add_argument('--unit', metavar='U', default='f',
+    p.add_argument('--unit-temperature', metavar='U', default='fahrenheit',
+                   choices=['fahrenheit', 'celsius'],
                    help="unit for temperature")
-    p.add_argument('--update-interval', metavar='I', type=int, default=60*3,
-                   help="update interval in seconds")
+    p.add_argument('--unit-speed', metavar='U', default='miles_hour',
+                   choices=['miles_hour', 'meters_sec'],
+                   help="unit for wind speed")
+    p.add_argument('--update-interval', metavar='I', type=int, default=60*10,
+                   help="update interval in seconds (default: 10 minutes)")
     p.add_argument('--wrap-i3-status', action='store_true')
-    p.add_argument('--timeout', metavar='T', type=float, default=2,
-                   help="timeout on weather request")
```

Comment:
Does pyowm document its timeout behavior? Regardless, I don't think you should remove this, because even if pyowm does have solid timeout logic it's probably still a lot longer than users are willing to have their i3bar hang for, so we should still enforce our own timeout (at least until we write a better solution like fetching the weather asynchronously).

Reply:
It's not documented, but it appears to default to 2 seconds, which seems reasonable to me: https://github.com/csparpa/pyowm/search?q=API_AVAILABILITY_TIMEOUT&unscoped_q=API_AVAILABILITY_TIMEOUT.  We could presumably change this by providing our own config module; I didn't look into how that works but pyowm advertises the ability to inject your own configuration.

Regardless, I did plan to implement a better solution after this PR, like having the weather update in a background thread or event loop instead of calling synchronously from the main loop (assuming that thread wouldn't hold the GIL during whatever internal DNS syscall was hanging in #5).